scripts/test_e2e.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -ex
+
+export BROWSERSTACK_USERNAME=andrewwylde1
+
+rm -rf .dfx
+npm install
+
+dfx start --background
+dfx canister create --all
+dfx build
+dfx canister install --all
+
+CAN_JSON="{\"linkedup_assets\": \"$(dfx canister id linkedup_assets)\"}"
+
+# run e2e
+echo $CAN_JSON >cypress/plugins/canister_ids.json
+BrowserStackLocal --key $BROWSERSTACK_APIKEY --daemon start
+
+npm install
+echo Starting cypress
```

Comment:
This is probably best on stderr

```
1>&2 echo Starting cypress
```

Reply:
(same for anything that is just informative logs of things as they happen. Save stdout for things that may want to be piped to `jq` or `xargs` or something)

Reply:
Ahh I didn't know about that, really. What's the reasoning there?

Reply:
@codelemur Basically a unix pipe will forward stdout but not stderr.

So e.g. if you have a program that writes JSON to stdout, you can pipe it to `jq` (as you've used in here).

But what if your json-stdout-producing program wants to log some warnings along the way? It can write those to stderr. That way piping to jq will still work, and the end-user of the shell can see informative messages on stderr.

Beyond the utility of that, any further reasoning is closer to art than science. I highly recommend reading this book as very efficient way to learn the reasoning of the authors of UNIX conventions themselves. https://en.wikipedia.org/wiki/The_Unix_Programming_Environment